### ISSUE TITLE
Invalid WPS518 Found implicit enumerate() call outside for loop

### ISSUE THREAD

**Issue by vnmabus** [wemake-services/wemake-python-styleguide#1883]
# Bug report

## What's wrong

The following code generates a WPS518 violation

```python
combs = combinations(
    range(len(coords)),
    degree,
)
```

## How it should be

I think that it should not report an error because the alternative

```github
combs = combinations(
    i, _ for i in enumerate(coords),
    degree,
)
```
is more verbose and less performant.

**Comment by sobolevn** [comment]
Yes! Thanks a lot for the report. 

PRs are welcome!
